HeCBench: fpdc-hip encounters unimplemented hipFuncSetCacheConfig

As part of HeCBench analysis I had encountered a benchmark that invokes hipFuncSetCacheConfig  function which is not supported by ChipStar at the moment. Is it possible to support this ? 

HeCBench/fpdc-hip> make run
./main 255 8 100 2
GPU FP Compressor v2.2
Copyright 2011-2020 Texas State University
CHIP error [TID 157286] [1695346398.814827886] : hipErrorNotSupported (Called a function which is not implemented) in chipStar/src/CHIPBindings.cc:1730:hipFuncSetCacheConfig

CHIP error [TID 157286] [1695346398.815013577] : Caught Error: hipErrorNotSupported
^Cmake: *** [Makefile:54: run] Interrupt

--- a/src/CHIPBindings.cc
+++ b/src/CHIPBindings.cc
@@ -1727,8 +1727,8 @@ hipError_t hipFuncSetCacheConfig(const void *Func, hipFuncCache_t Cfg) {
   CHIPInitialize();
   NULLCHECK(Func);
 
-  UNIMPLEMENTED(hipErrorNotSupported);
-  // RETURN(hipSuccess);
+  // 'Cfg' is ignored for now - API documentation allows it.
+  RETURN(hipSuccess);
 
   CHIP_CATCH
 }

--- a/tests/runtime/CMakeLists.txt
+++ b/tests/runtime/CMakeLists.txt
@@ -84,3 +84,5 @@ add_hip_runtime_test(TestStructWithFnPtr.hip)
 add_hip_runtime_test(TestHipInit.hip)
 
 add_shell_test(TestRuntimeWarnings.bash)
+
+add_hip_runtime_test(TestAPIs.hip)

--- a/tests/runtime/TestAPIs.hip
+++ b/tests/runtime/TestAPIs.hip
@@ -1,0 +1,22 @@
+// Test various HIP API functions.
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+#include <hip/hip_runtime.h>
+#include <cassert>
+
+__global__ void k() {}
+
+int main() {
+  // Non-conformant from C++ spec PoV. Not sure if HIP supports this.
+  auto *KPtr = reinterpret_cast<const void *>(k);
+
+  for (auto Cfg : {hipFuncCachePreferNone, hipFuncCachePreferShared,
+                   hipFuncCachePreferL1, hipFuncCachePreferEqual}) {
+    // Preferences should not return error code.
+    assert(hipFuncSetCacheConfig(KPtr, Cfg) == hipSuccess);
+    assert(hipFuncSetCacheConfig(nullptr, Cfg) != hipSuccess);
+  }
+
+  return 0;
+}

